addon/webextension/background/main.js

```diff
@@ -97,6 +97,15 @@ window.main = (function () {
         title: browser.i18n.getMessage("notificationLinkCopiedTitle"),
         message: browser.i18n.getMessage("notificationLinkCopiedDetails", pasteSymbol)
       });
+    } else {
+      catcher.watchPromise(browser.tabs.query({active: true, currentWindow: true})
+      .then((tabs) => {
+        for (let tab of tabs) {
+          if(tab.url.match(/\/creating/i)) {
+            catcher.watchPromise(browser.tabs.remove(tab.id));
```

Comment:
@niharikak101 this is good but I think there's a way we can do this without looking through all the tabs. 

Look at this section of code: https://github.com/mozilla-services/screenshots/blob/dd75b421a70c459d7c69ca98590c746587beb682/addon/webextension/background/takeshot.js#L41-L51

We can add another function to the `then` after `uploadShot` to catch any errors and remove the tab. Handling the error close to where it happens is usually a good practice. In this case we still have a reference to the tab we opened, `openedTab`.